lib/extended-css.js

```diff
@@ -27,30 +27,80 @@ var ExtendedCss = function (styleSheet) { // jshint ignore:line
     var affectedElements = [];
     var domObserved;
 
-    /**
-     * Removes specified suffix from the string
-     */
-    var removeSuffix = function(str, suffix) {
+    var eventListenerSupported = window.addEventListener;
 
-        var index = str.indexOf(suffix, str.length - suffix.length);
-        if (index >= 0) {
-            return str.substring(0, index);
+    var domMutationObserver;
+
+    var observeDocument = function (callback) {
+        if (utils.MutationObserver) {
+            domMutationObserver = new utils.MutationObserver(function(mutations) {
+                if (mutations && mutations.length) {
+                    callback();
+                }
+            });
+            domMutationObserver.observe(document.documentElement, {
+                childList: true,
+                subtree: true,
+                attributes: false
+            });
+        } else if (eventListenerSupported) {
+            document.addEventListener('DOMNodeInserted', callback, false);
+            document.addEventListener('DOMNodeRemoved', callback, false);
+            document.addEventListener('DOMAttrModified', callback, false);
+        }
+    };
+    var disconnectDocument = function (callback) {
+        if (domMutationObserver) {
+            domMutationObserver.disconnect();
+        } else if (eventListenerSupported) {
+            document.removeEventListener('DOMNodeInserted', callback, false);
+            document.removeEventListener('DOMNodeRemoved', callback, false);
+            document.removeEventListener('DOMAttrModified', callback, false);
         }
+    };
 
-        return str;
+    var protectionObserverOption = {
+        attributes: true,
+        attributeOldValue: true,
+        attributeFilter: ['style']
+    };
+    var protectionFunction = function (mutations, observer) {
+        if (!mutations.length) { return; }
+        var target = mutations[0].target;
+        observer.disconnect();
+        for (let mutation of mutations) {
+            if (mutation.attributeName === 'style') {
+                target.setAttribute('style', mutation.oldValue);
+            }
+        }
+        observer.observe(target, protectionObserverOption);
+    };
+     /**
+     * Sets up a MutationObserver which protects style attributes from changes
+     * @param node DOM node
+     * @returns Mutation observer used to protect attribute or null if there's nothing to protect
+     */
+    var protectStyleAttribute = function (node) {
+        if (!utils.MutationObserver) { return null; }
+        var protectionObserver = new utils.MutationObserver(protectionFunction);
+        protectionObserver.observe(node, protectionObserverOption);
+        return protectionObserver;
     };
 
+    /** Should be clear what it is. */
```

Comment:
Make sure this is the last time you're making your point via the production code. This is childish, dangerous if tolerated, and unexpected of you.

Reply:
You seem to be unable to refrain from posting such a silly comment publicly, here is what I propose.
 - Stop discussing pull requests here in GitHub, and move all reviewing stuff to Bitbucket as we do with the browser extension.